mne/forward/_make_forward.py

```diff
@@ -934,3 +944,77 @@ def use_coil_def(fname):
         yield
     finally:
         _extra_coil_def_fname = None
+
+
+class _ForwardModeler:
+    """Optimized incremental fitting using the same sensors and BEM."""
+
+    @verbose
+    def __init__(
+        self,
+        info,
+        trans,
+        bem,
+        *,
+        mindist=0.0,
+        n_jobs=1,
+        verbose=None,
+    ):
+        self.mri_head_t, _ = _get_trans(trans)
+        self.mindist = mindist
+        self.n_jobs = n_jobs
+        src = SourceSpaces([])
+        self.sensors, _, _, _, self.bem = _prepare_for_forward(
+            src,
+            self.mri_head_t,
+            info,
+            bem,
+            mindist,
+            n_jobs,
+            bem_extra="",
+            trans="",
+            info_extra="",
+            meg=True,
+            eeg=True,
+            ignore_ref=False,
+        )
+        self.fwd_data = _prep_field_computation(
+            sensors=self.sensors,
+            bem=self.bem,
+            n_jobs=self.n_jobs,
+        )
+        if self.bem["is_sphere"]:
+            self.check_inside = _CheckInsideSphere(self.bem)
+        else:
+            self.check_inside = _CheckInside(_bem_find_surface(self.bem, "inner_skull"))
+
+    def compute(self, src):
+        src = _ensure_src(src).copy()
+        for s in src:
+            transform_surface_to(s, "head", self.mri_head_t)
```

Comment:
I see some lines in this method (such as these) that look similar/identical to stuff happening in `_prepare_for_forward`.  I'm assuming you've already done what you can to DRY, and there's just enough that is different to make it hard/impossible to abstract out a helper func.  But mentioning it here in case you missed the redundancy / just forgot to go back and refactor those bits.

Reply:
Okay made it a little more DRY by adding a `src._transform_to(...)` at least! There were maybe half a dozen places it turns out we used that `for s in src: transform_surface_to(s, ...)` pattern